smithy-openapi/src/main/java/software/amazon/smithy/openapi/fromsmithy/protocols/AbstractRestProtocol.java

```diff
@@ -138,7 +138,11 @@ abstract Schema createDocumentSchema(
 
             // Greedy labels in OpenAPI need to include the label in the generated parameter.
             // For example, given "/{foo+}", the parameter name must be "foo+".
-            String name = label.isGreedyLabel() ? label.getContent() + "+" : label.getContent();
+            // Some vendors/tooling, require the "+" suffix be excluded in the generated parameter.
+            // If required, the setRemoveGreedyParameterSuffix config option should be set to `true`.
+            // When this option is enabled, given "/{foo+}", the parameter name will be "foo".
+            String name = (label.isGreedyLabel() && !context.getConfig().getRemoveGreedyParameterSuffix())
```

Comment:
This would probably work better as a real `if` statement now that the ternary is so complex.

Reply:
Addressed in ee69c1c.